### PR TITLE
Cloud function

### PR DESCRIPTION
**Pull request by michelaburns**
Hey Rabab! I'm just making this a pull request so it's easier to find/see the code

### CODE REVIEW

File: function/index.js [hunk 1]
```diff
@@ -36,7 +51,7 @@ exports.predictImage = (req, res) => {
     score: prediction.score,
   }
   
-  res.status(200).json(result);
+  res.status(200).json(result).send(result.score);
```
Comment (elithrar) [inline function/index.js]: You’ll want to send the entire result object here, so you don’t need to add the `.send()`

File: function/index.js [hunk 2]
```diff
@@ -19,14 +19,29 @@ exports.predictImage = (req, res) => {
...
+  const filePath = "gs://sp19-codeu-35-7727-vcm/dog_dataset_two/";
+  const scoreThreshold = 0.5;
   
   // https://cloud.google.com/nodejs/docs/reference/automl/0.1.x/v1beta1.PredictionServiceClient#predict
-  const formattedName = client.modelPath('[PROJECT]', '[LOCATION]', '[MODEL]');
+  const formattedName = client.modelPath('sp19-codeu-35-7727', 'gs://sp19-codeu-35-7727-vcm/dog_dataset_two', 'dog_dataset_two');
   const payload = {};
+
+  const params = {};
+  if (scoreThreshold) {
```
Comment (elithrar) [inline function/index.js]: You can elide this. Let the consumer (the web app) determine whether it’s useful to provide a cut-off.

File: function/index.js [hunk 3]
```diff
@@ -19,14 +19,29 @@ exports.predictImage = (req, res) => {
   // 1. Take image from request (req) and POST to Auto ML predict API
   // 2. Handle errors - e.g. the image couldn't be predicted? No label returned?
   // 3. Return the Auto ML label / score back in the response via res.send
```
Comment (elithrar) [inline function/index.js]: Put these constants at the top of the file - above the function export line. 

Reason: they don’t need to be initialized on each request; just once.

**Review by elithrar — CHANGES_REQUESTED** [review]
A few comments. Outstanding questions:

- Did you test this by deploying it via the Cloud Functions section in the console?
- To test this without the front-end, use your terminal:

e.g. https://stackoverflow.com/questions/12667797/using-curl-to-upload-post-data-with-files

```sh
curl -XPOST $functionurlhere -F "imageBytes=@myimage.jpg"
```

Should return a result / status code.

File: function/index.js [hunk 4]
```diff
@@ -12,7 +12,8 @@ const client = new automl.PredictionServiceClient();
  * @param {Object} res Cloud Function response context.
  *                     More info: https://expressjs.com/en/api.html#res
  */
-exports.predictImage = (req, res) => {
+
+@@ -19,14 +19,29 @@ exports.predictImage = (req, res) => {
```
Comment (elithrar) [inline function/index.js:16]: Right after this line, add:

```
if (req.method === "OPTIONS") {
      res.set("Access-Control-Allow-Origin", "*")
      res.set("Access-Control-Allow-Headers", "Accept, Accept-Language, Content-Language, Content-Type")
      res.status(200).send()
    }
  
  	if (req.method !== "POST") {
      res.status(405).send()
    }
```

The browser needs this to send requests to a domain (the Cloud Functions one) different from where it’s hosted. 

Copied from the existing function: https://console.cloud.google.com/functions/details/us-central1/predict?project=sp19-codeu-35-7727&tab=source&duration=PT1H

**Review by elithrar — CHANGES_REQUESTED** [review]
Add this before the final existing `res.status` call:

```
res.set("Access-Control-Allow-Origin", "*")
```

Again, matching https://console.cloud.google.com/functions/details/us-central1/predict?project=sp19-codeu-35-7727&tab=source&duration=PT1H

**Comment by michelaburns** [comment]
Hey Rabab! I added the changes Matt mentioned above. See: https://github.com/codeu-spring2019/whats-that-doggo/pull/10

**Pull request closed without merging**